src/main/scala/viper/gobra/ast/internal/Program.scala

```diff
@@ -1070,8 +1147,15 @@ case class InterfaceT(name: String, addressability: Addressability) extends Type
     InterfaceT(name, newAddressability)
 }
 
+case class ChannelT(elem: Type, addressability: Addressability) extends Type {
```

Comment:
Correct, the type checker ensures that you only access the channel in permitted ways but for the encoding we can forget about it. It even simplifies the encoding because there should be an equality between the `Closed` mpredicate for all directions of a particular channel (put differently, the `Closed` mpredicate should take the supertype of all channel directions as receiver) (this is what @MartinClochard calls "Only if c bidirectional , otherwise type error" in the concurrency outline). Martin then recommended me to unify all channel directions instead of adding a top type.